docs/hands-on-learning/introduction/module-6-testing-recipes.md

```diff
@@ -0,0 +1,55 @@
+
+* Understand how to write unit tests for your recipes.
+* Learn how to assert the state of the LST before and after running a recipe.
+* Explore the various ways to provide recipe and source specifications.
+
+### Steps
+
+1. Open [src/test/java/com/yourorg/AppendToReleaseNotesTest.java](https://github.com/moderneinc/rewrite-recipe-starter/blob/main/src/test/java/com/yourorg/AppendToReleaseNotesTest.java).
+   * Notice how the recipe specification directly constructs a `JavaRecipe` and passes that in. This is most convenient when testing imperative recipes.
+   * Notice how `@Test void createNewReleaseNotes() { ... }` uses `org.openrewrite.test.SourceSpecs.text(java.lang.String, java.lang.String, ...)` to provide a before and after text block, with the third parameter using `spec -> spec.path(Path.of("RELEASE.md")` to set the path where the source file either exists or should be created.
+   * The before text block is `null` to indicate that the file does not exist initially. Conversely, you can pass in `null` as the second argument to indicate that the file should be deleted.
+   * `@Test void editExistingReleaseNotes()` also uses `spec -> spec.path(Paths.get("RELEASE.md")` to set the source file, such that the recipe will match.
+2. Open [src/test/java/com/yourorg/AssertEqualsToAssertThatTest.java](https://github.com/moderneinc/rewrite-recipe-starter/blob/main/src/test/java/com/yourorg/AssertEqualsToAssertThatTest.java).
+   * Note how `.parser(JavaParser.fromJavaVersion().classpath("junit-jupiter-api"))` is called on the recipe specification.
+   * Comment out just the `.classpath("junit-jupiter-api")` part and then run the test.
+   * The resulting `java.lang.IllegalStateException: LST contains missing or invalid type information` indicates that the type information is missing, and that the test classpath is likely not correctly set up.
+3. Open [src/test/java/com/yourorg/NoGuavaListsNewArrayListTest.java](https://github.com/moderneinc/rewrite-recipe-starter/blob/main/src/test/java/com/yourorg/NoGuavaListsNewArrayListTest.java).
+   * Read the various comments throughout this test class.
+   * Try to make minimal changes in the recipe and see how they affect the tests.
+4. Open [src/test/java/com/yourorg/ClassHierarchyTest.java](https://github.com/moderneinc/rewrite-recipe-starter/blob/main/src/test/java/com/yourorg/ClassHierarchyTest.java).
+   * Note how each `rewriteRun` consumes a `RecipeSpec` to assert the `dataTable` rows produced in the recipe run.
+   * Correlate this to the `insertRow` calls in the recipe, to see how the recipe produces the expected output.
+5. Open [src/test/java/com/yourorg/SimplifyTernaryTest.java](https://github.com/moderneinc/rewrite-recipe-starter/blob/main/src/test/java/com/yourorg/SimplifyTernaryTest.java).
+   * Note how the instantiated recipe is a generated class, not the Refaster template class itself. We'll take a closer look at Refaster templates in the [next module](module-6-refaster-recipes).
```

Comment:
This link is wrong. It should start with `./` and also this current one is module 6 -- so the next module should point to module 7.